src/index.ts

```diff
@@ -200,11 +259,29 @@ export class Amplitude {
    * Send an identify call containing user property operations to Amplitude servers.
    *
    * @param identifyInstance
+   * @param extra
    */
-  identify(identifyInstance: Identify): Promise<boolean> {
-    return AmplitudeReactNative.identify(
-      this.instanceName,
-      identifyInstance.payload,
+  identify(
+    identifyInstance: Identify,
+    extra?: MiddlewareExtra,
+  ): Promise<boolean> {
+    return AmplitudeReactNative.getUserId(this.instanceName).then((userId) =>
+      AmplitudeReactNative.getDeviceId(this.instanceName).then((deviceId) => {
+        const event: IdentifyEvent = {
+          eventType: SpecialEventType.IDENTIFY,
+          userId,
+          deviceId,
+          userProperties: deepClonePayload(identifyInstance.payload),
```

Comment:
Even though it's safer, I'm a little concerned to run deepClonePayload() on every event. 

I think to minimize any side effects (extra processing, slowness) of middleware we should:
1. Don't call `getUserId()` or `getDeviceId()` prior to middleware. If the user needs the user_id they can call it explicitly in Middleware to get the current value
2. Pass shallow clone of properties instead of using JSON.tostrin().toJson()

cc @qingzhuozhen

Reply:
Agreed, sounds good to me.

Reply:
Fixed